lib/mongoid/indexable/specification.rb

```diff
@@ -93,9 +93,18 @@ def normalize_key(key)
       # @return [ Hash ] The normalized options.
       def normalize_options(opts)
         options = {}
+
         opts.each_pair do |option, value|
+          option = option.to_sym
           options[MAPPINGS[option] || option] = value
         end
+
+        %i[partial_filter_expression weights wildcard_projection].each do |key|
+          options[key] = normalize_key(options[key]) if options[key].is_a?(Hash)
+        end
+
+        options.dig(:partial_filter_expression, :'$and')&.map! { |expr| normalize_key(expr) }
```

Comment:
This looks to me like it would mutate user input.

Reply:
You're right. I've committed a fix.